arviz/plots/tsplot.py

```diff
@@ -0,0 +1,317 @@
+"""Plot timeseries data."""
+import warnings
+import numpy as np
+
+from ..sel_utils import xarray_var_iter
+from ..rcparams import rcParams
+from .plot_utils import default_grid, get_plotting_function
+
+
+def plot_ts(
+    idata,
+    y,
+    x=None,
+    y_hat=None,
+    y_holdout=None,
+    x_holdout=None,
+    y_forecasts=None,
+    plot_dim=None,
+    holdout_dim=None,
+    num_samples=100,
+    backend=None,
+    backend_kwargs=None,
+    y_kwargs=None,
+    y_hat_plot_kwargs=None,
+    y_mean_plot_kwargs=None,
+    vline_kwargs=None,
+    textsize=None,
+    figsize=None,
+    legend=True,
+    grid=False,
+    axes=None,
+    show=None,
+):
+    """Plot timeseries and it's components.
+
+    Parameters
+    ----------
+    idata : InferenceData
+    y : str, variable name from observed_data
+    x : str, Optional
+        If none, coords name of y dims
+    y_hat : str, from posterior_predictive, optional
+        Assumed to be of shape (chain, draw, *y.dims)
+    y_holdout : str, from observed_data, optional
+    x_holdout : str, from constant_data or y_holdout coords, optional
+    y_forecasts : str, from posterior_predictive, optional
+    plot_dim: str, Optional
+        Necessary if y is multidimensional.
+    holdout_dim: str, Optional
+        Necessary if y_holdout or y_forecasts is multidimensional.
+    num_samples : int, Optional, Default 100
+    backend : str, Optional
+        Select plotting backend {"matplotlib","bokeh"}. Default "matplotlib".
+    backend_kwargs : dict, optional
+    figsize : tuple, optional
+        Figure size. If None it will be defined automatically.
+    textsize : float, optional
+        Text size scaling factor for labels, titles and lines. If None it will be
+        autoscaled based on figsize.
+
+
+    Returns
+    -------
+    axes: matplotlib axes or bokeh figures.
```

Comment:
Quick initial comment, we should include examples somewhere, whether here or in the examples folder

Reply:
examples should go in both places ideally, not necessarily in this PR. Also, if you liked one model from ROS or another resource and used it as example for testing, it could be added as an example to ArviZ so it's available via `load_arviz_data`. Tests can't use examples that require internet connection but docs do. See #1132

Reply:
More general outside of this PR @utkarsh-maheshwari it would be good to have a medium length blog post explaining this and plot_lm using real world examples. Itll be a good exercise for you to practice technical communication in a more general audience voice, and be beneficial to the community. Importantly it will help us advertise your work so people are both well aware of it and how to use it!

Reply:
Great advice! We have 10 days left for GSoC, which I would like to dedicate to documentation, examples and blog posts. Also, I will be available even after GSoC to maintain the functions.